src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -911,16 +917,59 @@ private function createRequestMatcher(ContainerBuilder $container, string $path
             return $this->requestMatchers[$id];
         }
 
-        // only add arguments that are necessary
-        $arguments = [$path, $host, $methods, $ips, $attributes, null, $port];
-        while (\count($arguments) > 0 && !end($arguments)) {
-            array_pop($arguments);
+        $arguments = [];
+        if ($methods) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([MethodRequestMatcher::class, $methods]), MethodRequestMatcher::class)
+                ->setPublic(false)
+                ->setArguments([$methods])
+            ;
+            $arguments[] = new Reference($lid);
+        }
+
+        if ($path) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([PathRequestMatcher::class, $path]), PathRequestMatcher::class)
```

Comment:
maybe add a leading `.` to "hide" the services?

Reply:
`$id` already has a leading `.`